src/handlers/resources/ResourcesStreamHandler.cpp

```diff
@@ -0,0 +1,157 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "ResourcesStreamHandler.h"
+#include "ThreadMonitor.h"
+
+namespace visor::handler::resources {
+
+ResourcesStreamHandler::ResourcesStreamHandler(const std::string &name, InputStream *stream, const Configurable *window_config, StreamHandler *handler)
+    : visor::StreamMetricsHandler<ResourcesMetricsManager>(name, window_config)
+    , _timer(time(NULL))
+{
+    if (handler) {
+        throw StreamHandlerException(fmt::format("ResourcesStreamHandler: unsupported upstream chained stream handler {}", handler->name()));
+    }
+
+    assert(stream);
+    // figure out which input stream we have
+    if (stream) {
+        _pcap_stream = dynamic_cast<PcapInputStream *>(stream);
+        _dnstap_stream = dynamic_cast<DnstapInputStream *>(stream);
+        _mock_stream = dynamic_cast<MockInputStream *>(stream);
+        _sflow_stream = dynamic_cast<SflowInputStream *>(stream);
+        if (!_pcap_stream && !_mock_stream && !_dnstap_stream && !_sflow_stream) {
+            throw StreamHandlerException(fmt::format("NetStreamHandler: unsupported input stream {}", stream->name()));
+        }
+    }
+}
+
+void ResourcesStreamHandler::start()
+{
+    if (_running) {
+        return;
+    }
+
+    if (config_exists("recorded_stream")) {
+        _metrics->set_recorded_stream();
+    }
+
```

Comment:
The idea is to attach to the input callbacks that should be alway called to ensure that resource usage will be called sometimes

Reply:
User will decide when they want to attach this handler to their policy or not